### PR TITLE
Refactor dicom importer

### PR DESCRIPTION
**Pull request by Raedlr**
- object oriented structure, import handled by matRad_DicomImporter
- Adapted GUI to work with new Importer
- scan is part of the importer and output adapted for better compatibility

### CODE REVIEW

**Comment by wahln** [comment]
Thanks for the Pull Request!
@amitantony could you do the review?
Also, we should think about a way to run tests on the DICOM import / export. First step could be to take a very small dummy ct / cst / dose, export it with the dicom exporter, and then try to reimport those by script again.

File: matRad/dicom/@matRad_DicomImporter/matRad_createCst.m
Comment (Raedlr) [inline matRad/dicom/@matRad_DicomImporter/matRad_createCst.m:10]: Test

File: matRad/gui/widgets/matRad_importDicomWidget.m
Comment (amitantony) [inline matRad/gui/widgets/matRad_importDicomWidget.m:898]: can we rename this to a more descriptive name for the funtion and not just "scan" ? 
like "scanFolder"  or something

File: matRad/util/matRad_comparePlnStf.m
Comment (amitantony) [inline matRad/util/matRad_comparePlnStf.m:68]: change variable name to match matRad rules and could chnage it to a descriptive name "matchRadiationMode"

File: matRad/dicom/@matRad_DicomImporter/matRad_DicomImporter.m
Comment (amitantony) [inline matRad/dicom/@matRad_DicomImporter/matRad_DicomImporter.m:102]: clean up the comments 😋

File: matRad/dicom/matRad_interpDicomCtCube.m
Comment (amitantony) [inline matRad/dicom/matRad_interpDicomCtCube.m:13]: maybe change from 'doseGrid' to 'importGrid'

File: matRad/dicom/@matRad_DicomImporter/matRad_interpDicomDoseCube.m
Comment (amitantony) [inline matRad/dicom/@matRad_DicomImporter/matRad_interpDicomDoseCube.m:47]: can remove unused lines

File: matRad/dicom/@matRad_DicomImporter/matRad_interpDicomDoseCube.m
Comment (amitantony) [inline matRad/dicom/@matRad_DicomImporter/matRad_interpDicomDoseCube.m:50]: whats going on here ?

**Review by amitantony — COMMENTED** [review]
Overall the PR looks very good and clean 👍  I have given few format and minimal comments, will run some testing before final approval

File: matRad/dicom/@matRad_DicomImporter/matRad_interpDicomDoseCube.m
Comment (Raedlr) [reply matRad/dicom/@matRad_DicomImporter/matRad_interpDicomDoseCube.m:50]: I remember that we discussed this, and I still have no idea, what is this for

File: matRad/dicom/matRad_interpDicomCtCube.m
Comment (Raedlr) [reply matRad/dicom/matRad_interpDicomCtCube.m:13]: Was changed in this update https://github.com/e0404/matRad/pull/758/commits/c5b44adb21f8a5a279db925d26e087b3b0749f0f

File: matRad/dicom/@matRad_DicomImporter/matRad_importDicom.m
Comment (Raedlr) [reply matRad/dicom/@matRad_DicomImporter/matRad_importDicom.m]: I've changed the description here [c5b44ad](https://github.com/e0404/matRad/pull/758/commits/c5b44adb21f8a5a279db925d26e087b3b0749f0f)

File: matRad/gui/widgets/matRad_importDicomWidget.m
Comment (Raedlr) [reply matRad/gui/widgets/matRad_importDicomWidget.m]: renamed here [5dcd737](https://github.com/e0404/matRad/pull/758/commits/5dcd7378230d0a3fe9d0e7d6bbb2fbfd5e386e43)

**Comment by Raedlr** [comment]
matRad_importDicomSteeringParticles* in the last one

File: matRad/dicom/@matRad_DicomImporter/matRad_interpDicomDoseCube.m
Comment (amitantony) [reply matRad/dicom/@matRad_DicomImporter/matRad_interpDicomDoseCube.m:51]: id suggest removing it

**Comment by amitantony** [comment]
final testing required for importing particle therapy plan, then we are ready to go

**Comment by wahln** [comment]
I did a few changes:
- added helper function to use temporary directory in tests
- extended the tests to do an export and an actual reimport
- temorary directory used for dicom export and reimport tests
- removed GUI elements from folder scanning (I don't want to many boxes opened when we are using the exporter from script!)

In matRad_interpDicomCtCube, there still seems to be an issue in case of SliceThickness being a String in this line:
https://github.com/e0404/matRad/blob/2a3c89557a7393e90e96c4a3dcd9ea4cb14d5674/matRad/dicom/matRad_interpDicomCtCube.m#L90

If this only happens with the data exported from us, then it might either be an exporter error or an import error. The standard suggests a "Decimal String" for the attribute, so its unclear.

File: matRad/dicom/@matRad_DicomImporter/matRad_importDicomCt.m
Comment (wahln) [inline matRad/dicom/@matRad_DicomImporter/matRad_importDicomCt.m:68]: Are you certain that this is always a string / char array? Because if it is a number, it will become NaN here.

File: matRad/dicom/@matRad_DicomImporter/matRad_scanDicomImportFolder.m
Comment (wahln) [inline matRad/dicom/@matRad_DicomImporter/matRad_scanDicomImportFolder.m:295]: This is opening a dialog waiting for an answer. Dialogs should not be opened in the functions (which might also beused from scripts) but in the corresponding GUI piece.

**Comment by wahln** [comment]
Pushed some fixes to pass octave tests

**Review by wahln — APPROVED** [review]

**Pull request merged**
